salt/loader.py

```diff
@@ -1679,8 +1679,10 @@ def _load_module(self, name):
         ))
 
         for attr in getattr(mod, '__load__', dir(mod)):
-            if attr.startswith('_'):
-                # private functions are skipped
+            if attr.startswith('_') and attr != '__call__':
+                # private functions are skipped,
+                # except __call__ which is default entrance
+                # for multi-function batch-like state syntax
```

Comment:
This allows modules to have optional `__call__` function which is hidden in the module. So instead of:

```yaml
my_something_to_do:
  somemodule.function:
    - param

my_something_to_do_part_two:
  somemodule.otherfunction:
     - param

my_something_why_do_i_keep_doing_this:
  somemodule.thirdfunction:
    - param

my_something_oh_stop_that_please:
  somemodule.last_function:
    - param
```

You can now do it much simpler:
```yaml
my_something_to_do:
  somemodule:
    - function:
      - param
    - otherfunction:
      - param
    - last_function:
      - param

```
If you do not define function after `somemodule`, then in this case `__call__` will be injected to `somemodule.__call__` and it will iterate over the list (order preserved) introspecting module and calling things one by one.

Both syntaxes in parallel can perfectly exist.

Reply:
Every function with underscore is yanked from the module to get exported. Which means "private". I personally don't think this is the best solution, because it is impossible to keep hidden function that does not starts from underscore. In Zope, GvR made another convention: no `__doc__` = stays private. Anyway, only `__call__` is allowed. It (conceptually) works this way:

```python
def __call__(*function_names, **params):
    ret = []
    for func in function_names:
        ret.append(getattr(this_module, func)(*params[func]['args'], **params[func]['kwargs']))
    return ret
```